1091. Shortest Path in Binary Matrix/solution.py

```diff
@@ -4,23 +4,77 @@ class Solution:
     # cells containing a 0. Each cell is connected to the 8 cells around it
     # that either share an edge or a corner. If no path is found, return -1.
     def shortestPathBinaryMatrix(self, grid: List[List[int]]) -> int:
+        # Check that top left and bottom right cells contain 0s. Otherwise no clear
+        # path can exist, so return -1
+        gridHeight = len(grid)
+        gridWidth = len(grid[0])
```

Comment:
Sometimes reading the constraints is good. It said that n is both equal to the width of the grid and the height of the grid. In other words, it's a square matrix even though nothing else told us that it was a square matrix.

Reply:
Ooo I didn't notice that, thanks :) I'll keep this as is.